models/balances/polygon/erc20/balances_polygon_erc20_day.sql

```diff
@@ -52,12 +53,12 @@ FROM
 daily_balances b
 INNER JOIN 
 days d 
-    ON b.day <= d.day 
-    AND d.day < b.next_day
+    ON b.day <= d.block_day 
+    AND d.block_day < b.next_day
 LEFT JOIN 
 {{ source('prices', 'usd') }} p
     ON p.contract_address = b.token_address
-    AND d.day = p.minute
+    AND d.block_day = p.minute
```

Comment:
based on how the sequence of days is built above, would this `block_day` value always have a timestamp of `00:00` since its truncated to day? then if so, we always join to prices on that particular minute? is that better than aggregating minute in prices up to day and averaging?

Reply:
Exactly that's how it's working now!

Hmm that's a good question, I'm not sure what's better I think both are fair as long as it's transparent as we describe it in the table description. This join like this is also cheaper computationally probably.

I am temped to leave this as is right now, and if we think the "mean" approach makes sense we should create a separate spell for "mean daily price" and join on that everywhere we use this logic, rather than creating CTEs on each spell.

Reply:
What do you think @jeff-dude ?

Reply:
works for me, just raising awareness 👍 look to call that out in the readme specifically, that'll help be transparent and that we thought of both approaches

Reply:
Awesome @jeff-dude , thanks for thinking out loud with me on this